### DISCUSSION TITLE
Push "stable for diff" philosophy further

### DISCUSSION THREAD

**Discussion by zumm** [antfu/eslint-config#563] [Ideas]
Doc claims "stable for diff" as one of core styling principles and i see some room for improvement here.

1. At this moment config enforces no parens for arrow function with single argument. This produces noise in diff when argument count changes. Will be nice to get rid of this.
```
'style/arrow-parens': ['error', 'always'],
```
2. Enforcing "1 component prop/attr = 1 line" rule may greatly improve diff stability,
```
 'vue/max-attributes-per-line': ['error', { singleline: { max: 1 }, multiline: { max: 1 } }],
 'vue/first-attribute-linebreak': ['error', { singleline: 'beside', multiline: 'below' }],
```

@antfu What do you think about it? Sorry for ping, i know there is "But i prefer XXX" section in doc, but this is not about preferences, this is about following core styling principles.

**Comment by antfu** [comment]
Before "stable for diff" I prioritized a bit more for "minimal for reading". 

1. Where `arrow-parens` doesn't cause the diff issue to me, as when you change the function, the line of argument is already changed, which does not create extra line for diffing. In that sense, I'd prefer to avoid unnecessary parens as they could make the reading a bit harder.

2. I use UnoCSS https://unocss.dev/presets/attributify, where 1 attr per line is definitely not the style I want. Feel free to override the config on your own.

Thanks for bringing this up, and I hope I have answered your questions.

**Reply by zumm** [reply]
Thx for answer! But after this sentence:

>     2. I use UnoCSS https://unocss.dev/presets/attributify, where 1 attr per line is definitely not the style I want. Feel free to override the config on your own.

I'm tempted to ask how exactly do you use attributify? I'm using it too and i thought its purpose is to break long `class` line into shorter ones. No point to keep these attrs on same line. Do you use "valueless" variation? It's only case where forcing "1 attr per line" may be clunky i could imagine.

**Reply by antfu** [reply]
Yes I use valueless in mix of grouped attributify
